staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -107,7 +107,7 @@ func NewCmdWait(restClientGetter genericclioptions.RESTClientGetter, streams gen
 	flags := NewWaitFlags(restClientGetter, streams)
 
 	cmd := &cobra.Command{
-		Use:     "wait ([-f FILENAME] | resource.group/resource.name | resource.group [(-l label | --all)]) [--for=delete|--for condition=available]",
+		Use:     "wait ([-f FILENAME] | [-k DIRECTORY] | resource.group/resource.name | resource.group [(-l label | --all)]) [--for=delete|--for condition=available]",
```

Comment:
I see `|` implying that these options are exclusive, but I don't think there is anywhere that is validating that `-f` and `-k` are not used at the same time.

I see the flag description telling the user they `-f` and `-k` (and `-R`) cannot be used together:
> Process a kustomization directory. This flag can't be used together with -f or -R. 

... but I don't see any code preventing this combination.

https://github.com/kubernetes/kubernetes/blob/81ea5adff15e0baa70644ff07a92912cbd6cce29/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/filename_flags.go#L39-79

I realize this isn't something directly caused by your changes, so maybe this can just be a new issue for someone to add validation to prevent this invalid combination.  But if you want to address it here that is fine too.

Reply:
Actually there's already the validation: https://github.com/kubernetes/kubernetes/blob/7f744be14e2ff4823f77c6f36f6cdd518dbdde4b/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/builder_flags.go#L145.
Calling this function will run into `filenameOptions.validate()`.

Reply:
Yup, this was added when support for kustomize was added.

Reply:
OK, I see it now, in builder.go...